From 3125491b7c943a8ea6b89ea71eaf282f8575a9cc Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Mon, 17 May 2010 20:33:52 +0200
Subject: [PATCH 1/2] Do not hide that much

A bit more benevolent visibility. Needed on f13, still needs to be
looked into.

/usr/bin/ld: VirtualBox-3.2.0_OSE/out/linux.x86/release/obj/webtest/webtest:
hidden symbol `namespaces' in VirtualBox-3.2.0_OSE/out/linux.x86/release/obj/webtest/webtest.o
is referenced by DSO
/usr/bin/ld: final link failed: Nonrepresentable section on output
collect2: ld returned 1 exit status
---
 Config.kmk                                         |    2 +-
 src/VBox/Main/webservice/websrv-nsmap.xsl          |    2 ++
 2 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/Config.kmk b/Config.kmk
index d0c6e88..0cb1a93 100644
--- a/Config.kmk
+++ b/Config.kmk
@@ -2717,7 +2717,7 @@ ifeq ($(KBUILD_TARGET),win)
 
 else # the gcc guys
 TEMPLATE_VBOXR3EXE_TOOL                = $(VBOX_GCC_TOOL)
-TEMPLATE_VBOXR3EXE_CXXFLAGS            = -g -pipe $(VBOX_GCC_PEDANTIC_CXX) $(VBOX_GCC_Wno-variadic-macros) $(VBOX_GCC_OPT) $(VBOX_GCC_FP) -fno-strict-aliasing $(VBOX_GCC_fvisibility-hidden) $(VBOX_GCC_fvisibility-inlines-hidden)
+TEMPLATE_VBOXR3EXE_CXXFLAGS            = -g -pipe $(VBOX_GCC_PEDANTIC_CXX) $(VBOX_GCC_Wno-variadic-macros) $(VBOX_GCC_OPT) $(VBOX_GCC_FP) -fno-strict-aliasing $(VBOX_GCC_fvisibility-inlines-hidden)
 TEMPLATE_VBOXR3EXE_CXXFLAGS.x86        = -m32
 TEMPLATE_VBOXR3EXE_CXXFLAGS.amd64      = -m64
 TEMPLATE_VBOXR3EXE_CXXFLAGS.sparc32    = -m32
diff --git a/src/VBox/Main/webservice/websrv-nsmap.xsl b/src/VBox/Main/webservice/websrv-nsmap.xsl
index c515819..2a4de13 100644
--- a/src/VBox/Main/webservice/websrv-nsmap.xsl
+++ b/src/VBox/Main/webservice/websrv-nsmap.xsl
@@ -47,6 +47,7 @@
  * Generated from: src/VBox/Main/idl/VirtualBox.xidl (VirtualBox's interface definitions in XML)
  * Generator: src/VBox/Main/webservice/websrv-nsmap.xsl */
 
+#pragma GCC visibility push(default)
 #include "soapH.h"
 SOAP_NMAC struct Namespace namespaces[] =
 {
@@ -62,6 +63,7 @@ SOAP_NMAC struct Namespace namespaces[] =
   <xsl:text><![CDATA[
     {NULL, NULL, NULL, NULL}
 };
+#pragma GCC visibility pop
 
 ]]></xsl:text>
 </xsl:template>
-- 
1.7.0.1

